internal/cmd/auth.go

```diff
@@ -32,28 +39,70 @@ func authenticationGRPC(
 	cfg *config.Config,
 	forceMigrate bool,
 	authOpts ...containers.Option[auth.InterceptorOptions],
-) (grpcRegisterers, []grpc.UnaryServerInterceptor, func(context.Context) error, error) {
+) (grpcRegisterers, []grpc.UnaryServerInterceptor, []func(context.Context) error, error) {
 	// NOTE: we skip attempting to connect to any database in the situation that either the git or local
 	// FS backends are configured.
 	// All that is required to establish a connection for authentication is to either make auth required
 	// or configure at-least one authentication method (e.g. enable token method).
 	if !cfg.Authentication.Enabled() && (cfg.Storage.Type == config.GitStorageType || cfg.Storage.Type == config.LocalStorageType) {
 		return grpcRegisterers{
 			public.NewServer(logger, cfg.Authentication),
-			auth.NewServer(logger, memory.NewStore()),
-		}, nil, func(ctx context.Context) error { return nil }, nil
+			auth.NewServer(logger, storageauthmemory.NewStore()),
+		}, nil, nil, nil
 	}
 
-	_, builder, driver, shutdown, err := getDB(ctx, logger, cfg, forceMigrate)
+	_, builder, driver, dbShutdown, err := getDB(ctx, logger, cfg, forceMigrate)
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
+	var shutdownFuncs []func(context.Context) error
```

Comment:
i dont love all this tbh

one idea is that we pass in the `shutdown handler` to this method instead of returning an `[]shutdownFunc`?